src/xacro/__init__.py

```diff
@@ -362,12 +362,14 @@ def is_include(elt):
     return True
 
 
-def get_include_files(filename_spec, symbols):
+def get_include_files(filename_spec, symbols, optional=False):
     try:
         filename_spec = abs_filename_spec(eval_text(filename_spec, symbols))
     except XacroException as e:
         if e.exc and isinstance(e.exc, NameError) and symbols is None:
             raise XacroException('variable filename is supported with --inorder option only')
+        elif e.exc.strerror == "No such file or directory" and optional is True:
```

Comment:
checking for a specific exception type should be done via `isinstance()` as above.

Reply:
Done for the new single try-catch